### ISSUE TITLE
Location of basic pb tutorial not found until after monkeying with finger client tutorial

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2406]
|<img alt="arch_o_median's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| arch_o_median reported|
|-|-|
|Trac ID|trac#2406|
|Type|defect|
|Created|2007-01-17 05:56:39Z|

  I became very confused trying to learn about pb.  I attempted to find a tutorial on the topic and started the relatively advanced:
http://twistedmatrix.com/projects/core/documentation/howto/tutorial/pb.html

  before realizing I was totally out of my league and went back to find:

http://twistedmatrix.com/projects/core/documentation/howto/pb-intro.html

on the:

http://twistedmatrix.com/projects/core/documentation/howto/index.html

page.

Attachments:

* [finger-tutorial-preambles.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/412/412a33ae14746612317b013dc23213cd79b5f3f3/d1d940c1f6dc14e655b37f1f81ef3959f90fef83.patch) (6923 bytes) - added by jesstess on 2009-09-07 02:36:57Z - 
* [finger-tutorial-preambles2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/412/412a33ae14746612317b013dc23213cd79b5f3f3/c98e1a73f10b1058f1ae3cc5b77f36e327673950.patch) (9790 bytes) - added by jesstess on 2009-12-14 21:55:54Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__2406 2406
type__defect defect
reporter__arch_o_median arch_o_median
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__pb pb
keywords__documentation documentation
time__1169013399000000 1169013399000000
changetime__1260848129000000 1260848129000000
version__None None
owner__ 
cc__thijs cc__jesstess cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_1"></a>|@radix commented|
|-|-|

Could you be a bit more clear about what you believe the problem to be? And, perhaps, how it could be improved?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

Think we should close this one..

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

All our documentation should have a preamble that explains what _else_ you need to already understand.  The main purpose isn't to help lost people find their way to the right docs, but that's a useful side effect.

So, the pb part of the tutorial should point people at the pb docs.  Likewise, the web page should point them at the web docs, etc.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess removed owner|
|-|-|

I realize that there has been a lot of discussion about improving or scrapping the finger tutorial (eg [#10754](https://github.com/twisted/twisted/issues/10754)) over the last 4 years, but until that happens:

I also started going through the howtos in the order presented in the core howto index when I first started looking at Twisted and was turned off by the hordes of topics used in the finger tutorial that are explained later in the index or not even in the core documentation at all.

To that effect here's a patch that 1) explains on the first page of the finger tutorial how the tutorial can be used in conjunction with the rest of the documentation, and 2) provides links to the primary howtos for topics discussed in a given section of the tutorial at the beginning of that section.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra set owner to @jesstess|
|-|-|

You're mixing .html and .xhtml references in this patch, I think it should be .xhtml only?

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_6"></a>|@jesstess removed owner|
|-|-|

Gah, I really thought I'd checked that - thanks thijs. Patch updated accordingly.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @jesstess|
|-|-|

This mostly looks like a nice improvement to me.  Thanks.  There are a couple of things which bear a bit more consideration though, I think:

  1. A couple of the new links just go to the core howto documentation index - the ones in configuration.xhtml and pb.xhtml, in particular.  It would be nice to have these links be a bit more specific.  There's a plugin document (plugin.xhtml), a document for twistd plugins in particular (tap.xhtml).   I don't think any of the docs really talk about packaging, though.  Does this seem reasonable?
  1. Our doc toolchain has a somewhat significant, rather long-standing shortcoming: [[[#3438](https://github.com/twisted/twisted/issues/3438)](https://github.com/twisted/twisted/issues/3438)](https://github.com/twisted/twisted/issues/3438).  This is going to make the links to the non-core howtos break when the docs are actually deployed on the website.  I'm open to suggestions at how we can work-around this without fixing #3438 so that we can improve these docs as soon as possible, but it may be that we just have to resolve #3438 to move forward in this area.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_8"></a>|@jesstess removed owner|
|-|-|

I changed the links in pb.xhtml and configuration.xhtml to reflect exarkun's comments. This engendered the addition of anchors to the core howto's index.xhtml.

[#3438](https://github.com/twisted/twisted/issues/3438) is closed, so as far as I know the cross-project links aren't a problem anymore, although I don't have a great way to test this.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra set owner to @jesstess|
|-|-|

Thanks jesstess. I'm not sure if Lore can handle links that span multiple lines like:

```
<li><a href="application.xhtml">Using the Twisted Application 
Framework</a></li>
```

To make sure it won't break, could you put this in a branch and run the documentation builder on it? Other than that it looks good to me.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_10"></a>|@jesstess commented|
|-|-|

I'd already lore'd it without error and checked the output visually, so I'll go ahead and apply the patch.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_11"></a>|@jesstess set status to `closed`|
|-|-|

(In [27778]) Apply finger-tutorial-preambles2.patch

Author: jesstess
Reviewer: exarkun, thijs
Fixes: [#2406](https://github.com/twisted/twisted/issues/2406)

Add preambles to each section of the finger tutorial linking to the
parts of the core howto covered in that section.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
